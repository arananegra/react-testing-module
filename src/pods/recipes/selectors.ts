import { getRecipesSelector } from './store';
import { createSelector } from 'reselect';
import { mapToCollection } from 'common/mappers/collection';
import { mapRecipeFromApiToVm } from './recipes.mappers';

export const getRecipesVMSelector = createSelector(
  getRecipesSelector,
  recipes => mapToCollection(recipes, mapRecipeFromApiToVm)
);
