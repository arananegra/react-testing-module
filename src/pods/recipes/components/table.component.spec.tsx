import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { mockRecipes } from '../api/recipes.mock-data';
import { TableComponent } from './table.component';

describe('Table component tests', () => {
  it('should render a happy table with simple props', () => {
    // Arrange
    const props = {
      recipes: mockRecipes,
    };

    // Act
    const { getByTestId, asFragment } = render(<TableComponent {...props} />);
    const tableElement = getByTestId('mui-table');
    //Assert

    expect(tableElement).not.toBeUndefined();
    expect(asFragment()).toMatchSnapshot();
  });


  it('should call handleNavigation when row is clicked', () => {
    // Arrange
/*    const props = {
      recipes: mockRecipes,
    };

    // Act
    const { getByTestId, asFragment } = render(<TableComponent {...props} />);
    const tableElement = getByTestId('mui-table');
    const textFieldElement = getByTestId('mui-textfield');
    fireEvent.change(textFieldElement, { target: { value: 'John' } });
    //Assert

    expect(tableElement).not.toBeUndefined();
    expect(asFragment()).toMatchSnapshot();*/
  });
});
