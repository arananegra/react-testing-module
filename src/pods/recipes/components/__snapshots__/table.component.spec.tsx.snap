// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component tests should render a happy table with simple props 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
    data-testid="mui-table"
  >
    <table
      class="MuiTable-root makeStyles-table-2"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Ingredients
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root makeStyles-row-3"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="table-cell"
            scope="row"
          >
            Omelette
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            2 eggs , cheese , salt , black pepper
          </td>
        </tr>
        <tr
          class="MuiTableRow-root makeStyles-row-3"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="table-cell"
            scope="row"
          >
            Salad with tomatoes
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            salad , 2 tomatoes , 2 avocados , 1 tooth garlic , 1 onion , lemon juice , salt , pepper
          </td>
        </tr>
        <tr
          class="MuiTableRow-root makeStyles-row-3"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body"
            data-testid="table-cell"
            scope="row"
          >
            Spaghetti with tomato sauce
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            spaghetti , 1 beef , 1/4 teaspoon garlic powder , 1 onion , 3 tomatoes , 4 mushrooms , oregano , salt
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
